extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -708,6 +711,42 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 	public getRequiredPackagesForKernel(kernelName: string): PythonPkgDetails[] {
 		return this._requiredKernelPackages.get(kernelName) ?? [];
 	}
+
+	public get runningOnSaw(): boolean {
+		return this._runningOnSAW;
+	}
+
+	public async updateKernelSpecPaths(kernelsFolder: string): Promise<void> {
+		if (!this._runningOnSAW || this._kernelSpecsUpdated) {
+			return;
+		}
+		try {
+			let fileNames = await fs.readdir(kernelsFolder);
+			let filePaths = fileNames.map(name => path.join(kernelsFolder, name));
+			let fileStats = await Promise.all(filePaths.map(path => fs.stat(path)));
+			let folderPaths = filePaths.filter((value, index) => value && fileStats[index].isDirectory());
+			let kernelFiles = folderPaths.map(folder => path.join(folder, 'kernel.json'));
+
+			await Promise.all(kernelFiles.map(file => this.updateKernelSpecPath(file)));
+
+			this._kernelSpecsUpdated = true;
+		}
+		catch (error) {
+			throw error;
```

Comment:
If we're just rethrowing the error here, then we don't need these try-catch blocks. Same with the other method below this one.

Reply:
removed this one. the one below is required as while debugging the error was not bubbled up.

Reply:
We're only rethrowing the error here, not emitting any kind of message. If it wasn't being surfaced before, then this won't fix that. We shouldn't be swallowing the error either, since we want the notebook startup to fail if there's any issue with the kernel setup here.

Reply:
The error that emitted below gets displayed in the error notification e.g it showed the exact error " file not found ", while without this try catch it was giving some javascript error. This will help if any issues pop up while testing on SAW where we cannot debug stuff.

Reply:
I tested throwing an error from copyKernelsToSystemJupyterDirs() and it was correctly surfaced as an error notification without any of this redundant try-catch logic. Can you double check to confirm that? I was just calling "throw Error(...)" from inside the method (substitute error message as appropriate).

Reply:
I did not realize that I had added try catch to copyKernelsToSystemJupyterDirs which made this try catch redundant. removed this